ikarus/python/dirichletvalues/dirichletvalues.hh

```diff
@@ -57,54 +61,118 @@ void registerDirichletValues(pybind11::handle scope, pybind11::class_<DirichletV
-            };
-            self.fixBoundaryDOFs(lambda);
-          });
+  cls.def_property_readonly("container", &DirichletValues::container);
+  cls.def_property_readonly("size", &DirichletValues::size);
+  cls.def_property_readonly("fixedDOFsize", &DirichletValues::fixedDOFsize);
+  cls.def("isConstrained", [](DirichletValues& self, std::size_t i) -> bool { return self.isConstrained(i); });
 
-  cls.def("fixBoundaryDOFsUsingLocalView",
-          [](DirichletValues& self,
-             const std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int, LocalViewWrapper&)>& f) {
-            auto lambda = [&](BackendType& vec, int localIndex, LocalView& lv) {
-              auto lvWrapper = LocalViewWrapper(lv.globalBasis());
-              // this can be simplified when
-              // https://gitlab.dune-project.org/staging/dune-functions/-/merge_requests/418 becomes available
-              pybind11::object obj = pybind11::cast(lv.element());
-              lvWrapper.bind(obj);
-              f(vec.vector(), localIndex, lvWrapper);
-            };
-            self.fixBoundaryDOFs(lambda);
-          });
+  cls.def("fixIthDOF", [](DirichletValues& self, std::size_t i) { self.fixIthDOF(MultiIndex(std::array{i})); });
+
+  using FixBoundaryDOFsWithGlobalIndexFunction = std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int)>;
+  using FixBoundaryDOFsWithLocalViewFunction =
+      std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int, LocalViewWrapper&)>;
+  using FixBoundaryDOFsWithIntersectionFunction =
+      std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int, LocalViewWrapper&, const Intersection&)>;
 
   cls.def(
-      "fixBoundaryDOFsUsingLocalViewAndIntersection",
-      [](DirichletValues& self,
-         const std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int, LocalViewWrapper&, const Intersection&)>& f) {
-        auto lambda = [&](BackendType& vec, int localIndex, LocalView& lv, const Intersection& intersection) {
-          auto lvWrapper = LocalViewWrapper(lv.globalBasis());
-          // this can be simplified when
-          // https://gitlab.dune-project.org/staging/dune-functions/-/merge_requests/418 becomes available
-          pybind11::object obj = pybind11::cast(lv.element());
-          lvWrapper.bind(obj);
-          f(vec.vector(), localIndex, lvWrapper, intersection);
-        };
-        self.fixBoundaryDOFs(lambda);
-      });
+      "fixBoundaryDOFs",
+      [&](DirichletValues& self, const pybind11::function& functor) {
+        // Disambiguate by number of arguments, as casting doesn't properly work with functions
+        pybind11::module inspect_module = pybind11::module::import("inspect");
+        pybind11::object result         = inspect_module.attr("signature")(functor).attr("parameters");
+        size_t numParams                = pybind11::len(result);
+
+        if (numParams == 2) {
+          auto& function = functor.template cast<const FixBoundaryDOFsWithGlobalIndexFunction>();
```

Comment:
You are casting here to a type but you store a reference. Maybe removing the `&`?

Reply:
Ah yes originally i wanted to cast as a reference, but this didnt work